backend/src/main/scala/com/softwaremill/bootzooka/DependenciesFactory.scala

```diff
@@ -0,0 +1,67 @@
+package com.softwaremill.bootzooka
+
+import cats.data.NonEmptyList
+import cats.effect.{IO, Resource}
+import com.softwaremill.bootzooka.config.Config
+import com.softwaremill.bootzooka.email.{EmailConfig, EmailModel, EmailService}
+import com.softwaremill.bootzooka.email.sender.{DummyEmailSender, EmailSender, MailgunEmailSender, SmtpEmailSender}
+import com.softwaremill.bootzooka.http.{Http, HttpApi, HttpConfig}
+import com.softwaremill.bootzooka.infrastructure.{DB, DBConfig}
+import com.softwaremill.bootzooka.metrics.{MetricsApi, VersionApi}
+import com.softwaremill.bootzooka.passwordreset.{PasswordResetApi, PasswordResetAuthToken, PasswordResetCodeModel}
+import com.softwaremill.bootzooka.security.{ApiKeyAuthToken, ApiKeyModel}
+import com.softwaremill.bootzooka.user.UserApi
+import com.softwaremill.bootzooka.util.{Clock, DefaultIdGenerator, IdGenerator}
+import com.softwaremill.macwire.autocats.autowire
+import doobie.util.transactor.Transactor
+import io.prometheus.client.CollectorRegistry
+import sttp.client3.SttpBackend
+
+object DependenciesFactory {
+  private case class Modules(api: HttpApi, emailService: EmailService)
+
+  def resource(config: Config, sttpBackend: Resource[IO, SttpBackend[IO, Any]], xa: Resource[IO, Transactor[IO]], clock: Clock): Resource[IO, (HttpApi, EmailService)] = {
+    lazy val collectorRegistry: CollectorRegistry = CollectorRegistry.defaultRegistry
+    lazy val idGenerator: IdGenerator = DefaultIdGenerator
+
+    def buildHttpApi(http: Http, userApi: UserApi, passwordResetApi: PasswordResetApi, metricsApi: MetricsApi, versionApi: VersionApi, collectorRegistry: CollectorRegistry, cfg: HttpConfig) =
+      new HttpApi(
+        http,
+        userApi.endpoints concatNel passwordResetApi.endpoints,
+        NonEmptyList.of(metricsApi.metricsEndpoint, versionApi.versionEndpoint),
+        collectorRegistry,
+        cfg)
+
+    def buildApiKeyAuthToken(apiKeyModel: ApiKeyModel): ApiKeyAuthToken = new ApiKeyAuthToken(apiKeyModel)
+
+    def buildPasswordResetAuthToken(passwordResetCodeModel: PasswordResetCodeModel): PasswordResetAuthToken = new PasswordResetAuthToken(passwordResetCodeModel)
```

Comment:
seems like constructor invocation, can't we remove this?

Reply:
We use the created instance as `AuthTokenOps[PasswordResetCode]`, so for now unfortunately we can't, but once we implement [this one](https://github.com/softwaremill/macwire/issues/184#issuecomment-1009648688), we will be able to use it here instead.